pkgs/build-support/dlang/dub-to-nix/default.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  writeShellScriptBin,
+  coreutils,
+  gnugrep,
+  jq,
+  nix,
+}:
+
+let
+  binPath = lib.makeBinPath [
+    coreutils
+    gnugrep
+    jq
+    nix
+  ];
+in
+writeShellScriptBin "dub-to-nix" ''
+  set -euo pipefail
```

Comment:
Perhaps use writeShellApplication, it sets these by default and does shellcheck.

AFAICT, all this needs passed is the PATH, so this could probably be spun out into a separate file for easier editing.

Reply:
Much cleaner, thanks :)